### PR TITLE
Add jetpack_disable_no_verify_ssl_certs filter

### LINKED ISSUE #2145: Provide a filter or constant for disabling the SSL verification bypass
Bypassing SSL verification errors in `Jetpack_Client::_wp_remote_request` is sub-optimal, and it should be possible to disable this behavior in code, via a filter or a constant.

https://github.com/Automattic/jetpack/blob/bf36945f525a525af2495e059bb3b7a921f3c90c/class.jetpack-client.php#L119

### PR DESCRIPTION
**Pull request by joshbetz**
Provides a filter for anyone that wants to force `sslverify`.

Fixes #2145

### CODE REVIEW

File: class.jetpack-client.php [hunk 1]
```diff
@@ -134,6 +134,10 @@ public static function remote_request( $args, $body = null ) {
 	 * @return array|WP_Error WP HTTP response on success
 	 */
 	public static function _wp_remote_request( $url, $args, $set_fallback = false ) {
+		if ( apply_filters( 'jetpack_disable_no_verify_ssl_certs', false ) ) {
```
Comment (jeherve) [inline class.jetpack-client.php]: Could you add [some documentation](https://make.wordpress.org/core/handbook/inline-documentation-standards/php-documentation-standards/#4-hooks-actions-and-filters) before the filter? See [these examples](https://github.com/Automattic/jetpack/pulls?utf8=%E2%9C%93&q=is%3Apr+label%3Adocs+).

**Comment by joshbetz** [comment]
Note: This effectively forces `sslverify`, though it's not completely intuitive the way this is written. We could probably make it more obvious by moving the filter, but it wouldn't have the advantage of skipping the option check. I'm hoping the documentation is sufficient.

**Comment by simonwheatley** [comment]
@joshbetz The double negative in the filter name really hurts my brain, do you think we could rename to:

```php
if ( apply_filters( 'jetpack_verify_ssl_certs', false ) ) {
```

Do we need to be more specific with the filter name, perhaps `jetpack_client_verify_ssl_certs`?

**Comment by joshbetz** [comment]
@simonwheatley I like that. @jeherve or @georgestephanis thoughts on changing the name of the filter to `jetpack_verify_ssl_certs`?

**Comment by jeherve** [comment]
Go for it. `jetpack_client_verify_ssl_certs` might be even better, as @simonwheatley suggested.

**Comment by simonwheatley** [comment]
@jeherve Are we good to go here?

**Comment by jeherve** [comment]
Seems good to me. @dereksmart and @zinigor handle this release, so they'll review and merge in the next few days!

**Comment by zinigor** [comment]
Looks good to me! I wanted to suggest further renaming the filter to include the word `force`, but then I thought that people who need it will see the docs, so no need for that.

**Pull request merged**